dashboard/app/controllers/omniauth_callbacks_controller.rb

```diff
@@ -309,15 +305,28 @@ def login(new_sign_up_url = '')
     end
   end
 
-  private def register_new_user(user, new_sign_up_url = '')
+  private def register_new_user(user)
     PartialRegistration.persist_attributes(session, user)
 
     @form_data = {
       email: user.email
     }
+    new_sign_up_url = determine_sign_up_url(user)
     render 'omniauth/redirect', layout: false, locals: {new_sign_up_url: new_sign_up_url}
   end
 
+  private def determine_sign_up_url(user)
+    user_type = cookies['new_sign_up_user_type']
+    cookies.delete('new_sign_up_user_type')
```

Comment:
I double checked with copilot- if the cookie doesn't exist, it will just return nil (and then end up in that third case!). Similarly, the delete call will look for and attempt to delete the cookie, but if it doesn't exist, won't throw an error.

Reply:
Okay perfect!